### DISCUSSION TITLE
ESP-IDF v5.x int32_t and uint32_t for Xtensa compiler changed

### DISCUSSION THREAD

**Discussion by dizcza** [UncleRus/esp-idf-lib#441] [General]
> The types int32_t and uint32_t have been changed from int and unsigned int to long and unsigned long.

See more at https://docs.espressif.com/projects/esp-idf/en/latest/esp32/migration-guides/release-5.x/gcc.html

I got errors compiling esp-idf-lib components. Because of old print formatting. For example, `i2cdev.c`:

```
  233 |     ESP_LOGD(TAG, "Timeout: ticks = %d (%d usec) on port %d", dev->timeout_ticks, dev->timeout_ticks / 80, dev->port);
```

For now, I remove the abusive lines manually. Care should be taken to go through all these print lines.

**Comment by UncleRus** [comment]
Fixed in latest `master` and 0.9.1 release.
